analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Operations.cs

```diff
@@ -294,7 +319,7 @@ public void ObjectCreation_SetsNotNull()
             validator.ValidateContainsOperation(OperationKind.ObjectCreation);
             validator.ValidateTag("Declared", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue());
             validator.ValidateTag("Assigned", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue());
-            validator.ValidateTag("ValueType", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue());
+            validator.ValidateTag("ValueType", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue());   // This is questionable, value types should not have ObjectConstraint
```

Comment:
> // This is questionable, value types should not have ObjectConstraint
I think ObjectConstraint should be settable to
* Reference types
* Nullable value types
* `class` constraint generic types
* Unconstraint generic types.

Reply:
For nullable value types - this needs to be done differently in MMF-2401. It's not `Null` constraint on the item itself (because `item.HasValue` should not report null pointer dereference) but rather `HasNullValue`